### PR TITLE
[DCOS-53756] StatsD Reporter unit tests

### PR DESCRIPTION
**Pull request by rpalaznik**
## What changes were proposed in this pull request?

Resolves [DCOS-53756](https://jira.mesosphere.com/browse/DCOS-53756)

- Adds unit tests for spark statsd reporter

## How were these changes tested?

Unit tests were executed locally.

## Release Notes

n/a

### CODE REVIEW

File: spark-statsd-reporter/src/test/java/org/apache/spark/metrics/sink/statsd/MetricFormatterTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+import static junit.framework.TestCase.assertTrue;
+import static org.apache.spark.metrics.sink.statsd.MetricType.GAUGE;
+
+
+
+public class MetricFormatterTest {
+
+    @Test
+    public void simpleDriverMetricString() {
```
Comment (samvantran) [inline spark-statsd-reporter/src/test/java/org/apache/spark/metrics/sink/statsd/MetricFormatterTest.java]: This all looks great but I don't think we're actually running unit tests for spark-build?

Comment (rpalaznik) [reply spark-statsd-reporter/src/test/java/org/apache/spark/metrics/sink/statsd/MetricFormatterTest.java]: Looks like they should be running, there is a target for it in the Makefile

https://github.com/mesosphere/spark-build/blob/master/Makefile#L47

File: spark-statsd-reporter/src/test/java/org/apache/spark/metrics/sink/statsd/InstanceDetailsProviderMock.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package org.apache.spark.metrics.sink.statsd;
+
+import java.util.Optional;
+
+public class InstanceDetailsProviderMock extends InstanceDetailsProvider {
```
Comment (akirillov) [inline spark-statsd-reporter/src/test/java/org/apache/spark/metrics/sink/statsd/InstanceDetailsProviderMock.java]: any reasons for not using Mockito instead?

File: spark-statsd-reporter/src/test/java/org/apache/spark/metrics/sink/statsd/MetricFormatterTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        InstanceDetailsProvider instanceDetailsProviderMock = new InstanceDetailsProviderMock(
+                "834d77b5-a7b1-4c9d-9742-0f95d39d15e0-0002-driver-20190430095449-0001", "Test Spark App",
+                InstanceType.DRIVER, "aa31a823-2c7c-40e0-aef9-4b2a42adb461", "default");
```
Comment (akirillov) [inline spark-statsd-reporter/src/test/java/org/apache/spark/metrics/sink/statsd/MetricFormatterTest.java]: not relevant for the test correctness but for Driver instance type `instanceId` always equals to `applicationId`

**Comment by akirillov** [comment]
soft ping @rpalaznik, let's avoid having dangling PRs.

File: spark-statsd-reporter/src/test/java/org/apache/spark/metrics/sink/statsd/InstanceDetailsProviderMock.java (hunk 2 above)
Comment (rpalaznik) [reply spark-statsd-reporter/src/test/java/org/apache/spark/metrics/sink/statsd/InstanceDetailsProviderMock.java]: Now that I used PowerMock for InstanceDetailsProvider tests, there is no reason not to use Mockito here as well. Changed as suggested.

File: spark-statsd-reporter/src/test/java/org/apache/spark/metrics/sink/statsd/MetricFormatterTest.java (hunk 3 above)
Comment (rpalaznik) [reply spark-statsd-reporter/src/test/java/org/apache/spark/metrics/sink/statsd/MetricFormatterTest.java]: Good to know, but I'll leave it as is, if it's not relevant for correctness.

**Pull request merged**